pkg/extension/adapter/discord.go

```diff
@@ -2,37 +2,53 @@
 package main
 
 import (
+	"fmt"
+	"github.com/bwmarrin/discordgo"
 	"github.com/hegemone/kore/pkg/comm"
-	"github.com/hegemone/kore/pkg/mock"
 	log "github.com/sirupsen/logrus"
+	"os"
 )
 
 type adapter struct {
-	client mock.PlatformClient
+	ingressChan chan<- comm.RawIngressMessage
+	client      *discordgo.Session
 }
 
 func (a *adapter) Name() string {
-	return "discord"
+	return "ex-discord.adapters.kore.nsk.io"
 }
 
 func (a *adapter) Listen(ingressCh chan<- comm.RawIngressMessage) {
```

Comment:
This needs tests.

Reply:
Looking back at this, I'm not sure any of these functions are good candidates to be tested. The functions from discordgo are already tested and the entire function revolves around those already tested functions.